examples/medplum-chart-demo/src/bots/core/charting-utils.ts

```diff
@@ -0,0 +1,281 @@
+ * Representing Diagnoses docs here: https://www.medplum.com/docs/charting/representing-diagnoses
+ *
+ * @param conditionData - Data object containg codes and values for a condition/reason for the visit.
+ * @param encounter - The encounter the data is derived from.
+ * @param user - The user creating the Condition resource
+ * @returns An array of bundle entries containing the Condition resource that can be created in a batch request.
+ */
+export function createConditions(conditionData: ConditionData, encounter: Encounter, user: Practitioner): Condition[] {
+  const conditions: Condition[] = [];
+  // const request: BundleEntryRequest = {
+  //   method: 'PUT',
+  //   url: `Condition?encounter=${getReferenceString(encounter)}&code=${code}`,
+  // };
+  // Create a condition for the encounter diagnosis
+  const encounterDiagnosis: Condition = {
+    resourceType: 'Condition',
+    subject: encounter.subject as Reference<Patient>,
+    encounter: { reference: getReferenceString(encounter) },
+    recorder: { reference: getReferenceString(user) },
+    asserter: { reference: getReferenceString(user) },
+    code: conditionData.reasonForVisit,
+    category: [
+      {
+        coding: [
+          {
+            system: 'http://hl7.org/fhir/ValueSet/condition-category',
+            code: 'encounter-diagnosis',
+            display: 'Encounter Diagnosis',
+          },
+        ],
+      },
+    ],
+  };
+
+  conditions.push(encounterDiagnosis);
+
+  // If the problem list question was checked, create an additional condition for it
+  if (conditionData.problemList) {
+    conditions.push({
+      resourceType: 'Condition',
+      subject: encounter.subject as Reference<Patient>,
+      encounter: { reference: getReferenceString(encounter) },
+      recorder: { reference: getReferenceString(user) },
+      asserter: { reference: getReferenceString(user) },
+      code: conditionData.reasonForVisit,
+      category: [
+        {
+          coding: [
+            {
+              system: 'http://hl7.org/fhir/ValueSet/condition-category',
+              code: 'problem-list-item',
+              display: 'Problem List Item',
+            },
+          ],
+        },
+      ],
+    });
+  }
+
+  return conditions;
+}
+
+/**
+ * This function takes ClinicalImpression data and creates a ClinicalImpression resource. The ClinicalImpression resource
+ * represents any notes on an encounter in this context.
+ *
+ * @param clinicalImpressionData - Data object containing codes and values for the ClinicalImpression resources
+ * @param encounter - The encounter that the data is derived from
+ * @param user - The user creating the ClinicalImpressions
+ * @returns A bundle entry with the ClinicalImpression resource that can be used in a batch transaction
+ */
+export function createClinicalImpressions(
+  clinicalImpressionData: ClinicalImpressionData,
```

Comment:
as with above, could `clinicalImpressionData` just be of type `Partial<ClinicalImpression>`, rather than a custom datatype?